integrations/Matomo/util.js

```diff
@@ -0,0 +1,332 @@
+            window._paq.push(["trackLink", url, linkType]);
+            break;
+
+          case "trackSiteSearch":
+            keyword =
+              get(properties, "keyword") || get(message, "context.page.search");
+            category = get(properties, "category");
+            resultsCount = get(properties, "resultsCount");
+            window._paq.push([
+              "trackSiteSearch",
+              keyword,
+              category,
+              resultsCount,
+            ]);
+            break;
+
+          case "ping":
+            window._paq.push(["ping"]);
+            break;
+
+          case "trackContentImpressionsWithinNode":
+            domId = get(properties, "domId") || get(properties, "dom_id");
+            window._paq.push([
+              "trackContentImpressionsWithinNode",
+              document.getElementById(domId),
+            ]);
+            break;
+
+          case "trackContentInteractionNode":
+            domId = get(properties, "domId") || get(properties, "dom_id");
+            contentInteraction = get(properties, "contentInteraction");
+            window._paq.push([
+              "trackContentInteractionNode",
+              document.getElementById(domId),
+              contentInteraction,
+            ]);
+            break;
+
+          case "trackContentImpression":
+            contentName = get(properties, "contentName");
+            contentPiece = get(properties, "contentPiece");
+            contentTarget = get(properties, "contentTarget");
+            window._paq.push([
+              "trackContentImpression",
+              contentName,
+              contentPiece,
+              contentTarget,
+            ]);
+            break;
+
+          case "trackContentInteraction":
+            contentInteraction = get(properties, "contentInteraction");
+            contentName = get(properties, "contentName");
+            contentPiece = get(properties, "contentPiece");
+            contentTarget = get(properties, "contentTarget");
+            window._paq.push([
+              "trackContentInteraction",
+              contentInteraction,
+              contentName,
+              contentPiece,
+              contentTarget,
+            ]);
+            break;
+
+          default:
+            break;
+        }
+      });
+    }
+  }, standardEventsMap);
+};
+
+/** Mapping Ecommerce Events
+ * @param  {} event
+ * @param  {} message
+ */
+const ecommerceEventsMapping = (event, message) => {
+  let productSKU;
+  let productName;
+  let categoryName;
+  let price;
+  let quantity;
+  let orderId;
+  let grandTotal;
+  let subTotal;
+  let tax;
+  let shipping;
+  let discount;
+  let category;
+  let action;
+  let name;
+  let value;
+  const products = get(message, "properties.products"); // used for TRACK_ECOMMERCE_ORDER event
+
+  switch (event) {
+    case "SET_ECOMMERCE_VIEW":
+      productSKU =
+        get(message, "properties.sku") || get(message, "properties.product_id");
```

Comment:
Same as above lets validate `properties` first then we can use . notation

Reply:
Or use optional chaining

Reply:
Done